File: samples/frontend/src/app/customer.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Injectable({ providedIn: 'root' })
export class CustomerService {
  private readonly base = '/api/customer';

  constructor(private http: HttpClient) {}

  all() {
    return this.http.get('/api/customer');
  }

  find(id: number) {
    return this.http.get(`/api/customer/${id}`);
  }

  add(customer: any) {
    return this.http.post('/api/customer', customer);
  }
}

File: samples/backend/Services/CustomerService.cs
using ShopApi.Models;

namespace ShopApi.Services;

public interface ICustomerService
{
    IEnumerable<Customer> All();
    Customer? Find(int id);
    void Add(Customer customer);
}

public class CustomerService : ICustomerService
{
    private readonly List<Customer> _customers = new();

    public IEnumerable<Customer> All() => _customers;
    public Customer? Find(int id) => _customers.FirstOrDefault(c => c.Id == id);
    public void Add(Customer customer) => _customers.Add(customer);
}
